frontend/src/container/BillingContainer/BillingContainer.test.tsx

```diff
@@ -0,0 +1,213 @@
+import { billingSuccessResponse } from 'mocks-server/__mockdata__/billing';
+import {
+	notOfTrailResponse,
+	trialConvertedToSubscriptionResponse,
+} from 'mocks-server/__mockdata__/licenses';
+import { server } from 'mocks-server/server';
+import { rest } from 'msw';
+import { act, render, screen } from 'tests/test-utils';
+import { getFormattedDate } from 'utils/timeUtils';
+
+import BillingContainer from './BillingContainer';
+
+const lisenceUrl = 'http://localhost/api/v2/licenses';
+
+describe('BillingContainer', () => {
+	test('Component should render', async () => {
+		act(() => {
+			render(<BillingContainer />);
+		});
+		const unit = screen.getAllByText(/unit/i);
+		expect(unit[1]).toBeInTheDocument();
+		const dataInjection = screen.getByRole('columnheader', {
+			name: /data ingested/i,
+		});
+		expect(dataInjection).toBeInTheDocument();
+		const pricePerUnit = screen.getByRole('columnheader', {
+			name: /price per unit/i,
+		});
+		expect(pricePerUnit).toBeInTheDocument();
+		const cost = screen.getByRole('columnheader', {
+			name: /cost \(billing period to date\)/i,
+		});
+		expect(cost).toBeInTheDocument();
+
+		const total = screen.getByRole('cell', {
+			name: /total/i,
+		});
+		expect(total).toBeInTheDocument();
+
+		const manageBilling = screen.getByRole('button', {
+			name: /manage billing/i,
+		});
+		expect(manageBilling).toBeInTheDocument();
+
+		const dollar = screen.getByRole('cell', {
+			name: /\$0/i,
+		});
+		expect(dollar).toBeInTheDocument();
+
+		const currentBill = screen.getByRole('heading', {
+			name: /current bill total/i,
+		});
+		expect(currentBill).toBeInTheDocument();
+	});
+
+	test('OnTrail', async () => {
```

Comment:
All onTrail data tests can be combined into a single one.

Reply:
Done.